edx_rest_framework_extensions/auth/jwt/tests/test_authentication.py

```diff
@@ -751,6 +757,121 @@ def test_authenticate_jwt_and_no_session_and_set_request_user(self, mock_set_cus
+
+        # simulating email change
+        user.email = 'new@example.com'
+        user.save()  # pylint: disable=no-member
+
+        self.client.force_login(user)
+
+        response = self.client.get(reverse('authenticated-view'))
+
+        assert response.status_code == 200
+
+    @override_settings(
+        EDX_DRF_EXTENSIONS={
+            ENABLE_JWT_AND_LMS_USER_EMAIL_MATCH: True,
+            'JWT_PAYLOAD_USER_ATTRIBUTE_MAPPING': {},
+            'JWT_PAYLOAD_MERGEABLE_USER_ATTRIBUTES': []
+        },
+        MIDDLEWARE=(
+            'django.contrib.sessions.middleware.SessionMiddleware',
+            'django.contrib.auth.middleware.AuthenticationMiddleware',
+            'edx_rest_framework_extensions.auth.jwt.middleware.JwtAuthCookieMiddleware',
+        ),
+        ROOT_URLCONF='edx_rest_framework_extensions.auth.jwt.tests.test_authentication',
+    )
+    @mock.patch('edx_rest_framework_extensions.auth.jwt.authentication.set_custom_attribute')
+    def test_authenticate_user_lms_and_jwt_email_mismatch_failure(self, mock_set_custom_attribute):
+        """
+        Tests failure when lms and jwt user email do not match if ENABLE_JWT_AND_LMS_USER_EMAIL_MATCH is True.
+        """
+        user_email = 'old@example.com'
+        user = factories.UserFactory(email=user_email)
+        jwt_header_payload, jwt_signature = self._get_test_jwt_token_payload_and_signature(user=user)
+
+        # Cookie parts will be recombined by JwtAuthCookieMiddleware
+        self.client.cookies = SimpleCookie({
+            jwt_cookie_header_payload_name(): jwt_header_payload,
+            jwt_cookie_signature_name(): jwt_signature,
+        })
+
+        # simulating email change
+        user.email = 'new@example.com'
+        user.save()  # pylint: disable=no-member
+
+        self.client.force_login(user)
+
+        response = self.client.get(reverse('authenticated-view'))
+
+        assert response.status_code == 401
+        mock_set_custom_attribute.assert_any_call(
+            'jwt_auth_failed',
+            "Exception:JwtUserEmailMismatchError('Failing JWT authentication due to jwt user email mismatch with lms "
+            "user email.')"
+        )
+
+    @override_settings(
+        EDX_DRF_EXTENSIONS={
+            ENABLE_JWT_AND_LMS_USER_EMAIL_MATCH: True,
+            'JWT_PAYLOAD_USER_ATTRIBUTE_MAPPING': {},
+            'JWT_PAYLOAD_MERGEABLE_USER_ATTRIBUTES': []
+        },
+        MIDDLEWARE=(
+            'django.contrib.sessions.middleware.SessionMiddleware',
+            'django.contrib.auth.middleware.AuthenticationMiddleware',
+            'edx_rest_framework_extensions.auth.jwt.middleware.JwtAuthCookieMiddleware',
+        ),
+        ROOT_URLCONF='edx_rest_framework_extensions.auth.jwt.tests.test_authentication',
+    )
+    @mock.patch('edx_rest_framework_extensions.auth.jwt.authentication.set_custom_attribute')
+    def test_authenticate_user_lms_and_jwt_email_mismatch_success(self, mock_set_custom_attribute):
```

Comment:
[nit] Renaming to `test_authenticate_user_lms_and_jwt_email_match_success` would be slightly more clear.